packages/components/src/internal/util/Date.ts

```diff
@@ -810,3 +820,70 @@ export function isDateTimeInPast(date: Date | string | number, timezone?: string
     const zonedTime = fromZonedTime(date_, timezone_);
     return zonedTime.getTime() <= new Date().getTime();
 }
+
+/**
+ * Given a current non-us date format (DMY), assuming useMDYDateParsing is false, return accepted alternative parse formats.
+ * In order to be able to parse DMY dates, all possible combinations of MDY date and/or datetime formats need to be considered.
+ * To help narrow down the list of formats, an optional inputDateStr can be provided.
+ * @param currentFormat The configured field/container/folder date/datetime format
+ * @param inputDateStr An optional input date string to help narrow down the list of formats to test
+ */
+export function getAltNonUSParseFormats(
+    currentFormat: string,
+    inputDateStr?: string /*helps narrow down list of formats*/
+): string | string[] {
+    const { useMDYDateParsing } = getServerContext();
+    if (useMDYDateParsing || !currentFormat.toLowerCase().startsWith('d')) {
+        return currentFormat;
+    }
+    const dayPart = ['dd'];
+    const monthPart = ['MM', 'MMM', 'MMMM'];
+    const yearParts = ['yy', 'yyyy'];
+
+    let seperators = ['-', '/', ''];
```

Comment:
Should we include `.` and `' '` as valid separators (e.g., `'02.04.2025' or '02 04 2025'`)?